crates/shared/src/http_solver/model.rs

```diff
@@ -376,6 +376,18 @@ pub enum SolverRejectionReason {
     /// The solution doesn't have a positive score. Currently this can happen
     /// only if the objective value is negative.
     NonPositiveScore,
+
+    /// The solution has a score that is too high. This can happen if the
+    /// score is higher than the maximum score (surplus + fees)
+    #[serde(rename_all = "camelCase")]
+    TooHighScore {
+        #[serde(with = "u256_decimal")]
+        surplus: U256,
+        #[serde(with = "u256_decimal")]
+        fees: U256,
+        #[serde(with = "u256_decimal")]
+        max_score: U256,
```

Comment:
Should we maybe also add the submitted score here just for easier debugging?

Reply:
That's a nice suggestion.Added.